### PR TITLE
Changes parameter parsing to clap

### PR DESCRIPTION
**Pull request by pedrohjordao**
- Uses clap to parse parameters
- Removes of "allow" directive where they are not necessary
- Removes of unused variables

### CODE REVIEW

**Comment by pedrohjordao** [comment]
I'm looking into reworking the printf implementation (a lot of c style code and lack of tests), this is the first PR that I plan to make
Is there information somewhere with what still needs to be implemented for printf?

File: src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs [hunk 1]
```diff
@@ -43,27 +43,25 @@ impl Formatter for CninetyNineHexFloatf {
...
     // to separate slices. If no digits after decimal point,
     // assign 0
-    let (mut first_segment_raw, second_segment_raw) = match analysis.decimal_pos {
-        Some(pos) => (&str_in[..pos], &str_in[pos + 1..]),
-        None => (str_in, "0"),
-    };
-    if first_segment_raw.is_empty() {
-        first_segment_raw = "0";
-    }
+    //let (mut first_segment_raw, second_segment_raw) = match analysis.decimal_pos {
```
Comment (sylvestre) [inline src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs:58]: why comment this part?

Comment (pedrohjordao) [reply src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs:58]: Dead code
Commented so I could remove warnings about unused/unread variables

Comment (sylvestre) [reply src/uu/printf/src/tokenize/num_format/formatters/cninetyninehexfloatf.rs:58]: ok, just remove it then :)

We still have access to the history in git if we need to

**Comment by sylvestre** [comment]
thanks

**Pull request merged**